--- a/Source/Reasoner/Classifier/COptimizedKPSetClassTestingItem.h	2026-09-18 12:58:52.279853702 +0200
+++ b/Source/Reasoner/Classifier/COptimizedKPSetClassTestingItem.h	2026-09-18 12:58:52.279853702 +0200
@@ -22,7 +22,7 @@
 #define KONCLUDE_REASONER_CLASSIFIER_COPTIMIZEDKPSETCLASSTESTINGITEM_H
 
 // Libraries includes
-
+#include <atomic>
 
 // Namespace includes
 #include "ClassifierSettings.h"
@@ -177,13 +177,13 @@
 					cint64 mPredCounter;
 
 					// true, if the satisfiability testing for this concept has started
-					bool mSatTestOrdered;
+					std::atomic<bool> mSatTestOrdered;
 					// true, if the satisfiability testing has finished
-					bool mTestedSat;
+					std::atomic<bool> mTestedSat;
 					// true, if for a super concept the unsatisfiability has been discovered
-					bool mUnsatDerivated;
+					std::atomic<bool> mUnsatDerivated;
 					// true, if for the concept the satisfiability has been extracted from a completion graph
-					bool mSatDerivated;
+					std::atomic<bool> mSatDerivated;
 					// true of false if the concept is satisfiable respectively unsatisfiable
 					bool mSatTestedResult;
 
